Cargo.toml

```diff
@@ -22,14 +22,21 @@ license = "MIT"
 repository = "https://github.com/JojiiOfficial/bluetooth-serial-port-async"
 
 [features]
+default = ["async_std"]
 # Feature to disable any tests which rely on hardware availability
 # eg: tests which attempt to create a BtSocket.
 test_without_hardware = []
+async_std = ["dep:async-std"]
+tokio = ["dep:tokio"]
 
 [dependencies]
-async-std = "1.8.0"
+async-std = { version = "1.8.0", optional = true }
 enum_primitive = "0.1.1"
 libc = "0.2.81"
 mio = "0.6.0"
 nix = "0.19.1"
 itertools = "0.10.0"
+tokio = { version = "1", features=["net"], optional = true }
+
+[dev-dependencies]
+tokio = { version = "1", features = ["full"] }
```

Comment:
Do we really need all features?

Reply:
Ive reduced the features. It builds but I've been unable to test it as I don't have access to bluetooth hardware atm.

I assume since it builds it will run.